src/protocols/secure_channel/PASESession.cpp

```diff
@@ -117,75 +117,10 @@ void PASESession::DiscardExchange()
     }
 }
 
-CHIP_ERROR PASESession::Serialize(PASESessionSerialized & output)
-{
-    PASESessionSerializable serializable;
-    VerifyOrReturnError(BASE64_ENCODED_LEN(sizeof(serializable)) <= sizeof(output.inner), CHIP_ERROR_INVALID_ARGUMENT);
-
-    ReturnErrorOnFailure(ToSerializable(serializable));
-
-    uint16_t serializedLen = chip::Base64Encode(Uint8::to_const_uchar(reinterpret_cast<uint8_t *>(&serializable)),
-                                                static_cast<uint16_t>(sizeof(serializable)), Uint8::to_char(output.inner));
-    VerifyOrReturnError(serializedLen > 0, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(serializedLen < sizeof(output.inner), CHIP_ERROR_INVALID_ARGUMENT);
-    output.inner[serializedLen] = '\0';
-
-    return CHIP_NO_ERROR;
-}
-
-CHIP_ERROR PASESession::Deserialize(PASESessionSerialized & input)
-{
-    PASESessionSerializable serializable;
-    size_t maxlen            = BASE64_ENCODED_LEN(sizeof(serializable));
-    size_t len               = strnlen(Uint8::to_char(input.inner), maxlen);
-    uint16_t deserializedLen = 0;
-
-    VerifyOrReturnError(len < sizeof(PASESessionSerialized), CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(CanCastTo<uint16_t>(len), CHIP_ERROR_INVALID_ARGUMENT);
-
-    memset(&serializable, 0, sizeof(serializable));
-    deserializedLen =
-        Base64Decode(Uint8::to_const_char(input.inner), static_cast<uint16_t>(len), Uint8::to_uchar((uint8_t *) &serializable));
-
-    VerifyOrReturnError(deserializedLen > 0, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(deserializedLen <= sizeof(serializable), CHIP_ERROR_INVALID_ARGUMENT);
-
-    return FromSerializable(serializable);
-}
-
-CHIP_ERROR PASESession::ToSerializable(PASESessionSerializable & serializable)
-{
-    VerifyOrReturnError(CanCastTo<uint16_t>(mKeLen), CHIP_ERROR_INTERNAL);
-
-    memset(&serializable, 0, sizeof(serializable));
-    serializable.mKeLen           = static_cast<uint16_t>(mKeLen);
-    serializable.mPairingComplete = (mPairingComplete) ? 1 : 0;
-    serializable.mLocalSessionId  = GetLocalSessionId();
-    serializable.mPeerSessionId   = GetPeerSessionId();
-
-    memcpy(serializable.mKe, mKe, mKeLen);
-
-    return CHIP_NO_ERROR;
-}
-
-CHIP_ERROR PASESession::FromSerializable(const PASESessionSerializable & serializable)
-{
-    mPairingComplete = (serializable.mPairingComplete == 1);
-    mKeLen           = static_cast<size_t>(serializable.mKeLen);
-
-    VerifyOrReturnError(mKeLen <= sizeof(mKe), CHIP_ERROR_INVALID_ARGUMENT);
-    memset(mKe, 0, sizeof(mKe));
-    memcpy(mKe, serializable.mKe, mKeLen);
-
-    SetLocalSessionId(serializable.mLocalSessionId);
-    SetPeerSessionId(serializable.mPeerSessionId);
-
-    return CHIP_NO_ERROR;
-}
-
-CHIP_ERROR PASESession::Init(uint16_t mySessionId, uint32_t setupCode, SessionEstablishmentDelegate * delegate)
+CHIP_ERROR PASESession::Init(SessionHolder & secureSessionHolder, uint32_t setupCode, SessionEstablishmentDelegate * delegate)
```

Comment:
Similar to CASESession: should be using rvalues where semantically we're taking ownership of the session holder, various places in this file.

Reply:
As suggested by @mrjerryjohns , PASESession and CASESession are now doing the allocation themselves, so this doesn't need to be passed in the first place.